### ISSUE TITLE
adding to a function issue

### ISSUE THREAD

**Issue by robert-wallis** [robpike/ivy#8]
Not sure whats going on, but when I try this on Android:

> ((sqrt(5)+1.0)/2.0)==((1.0+sqrt(5))/2.0)
0

> (1.0+sqrt(5))/2.0
1.61803398875

> ((sqrt(5)+1.0)/2.0)
1.22474487139

It seems that if the left side is a constant, then it works, but if the left side is a function then it doesn't.   Am I using it wrong?

**Comment by robpike** [comment]
Parentheses are not needed for function application. Your expression sqrt(5)+1.0 is exactly the same, parsewise, as sqrt 5 + 1.0, or sqrt 6.

You might like to turn on
)debug parse
to help you understand how it parses. It's APL's rules, and you get used to them after a while. :)

**Comment by robert-wallis** [comment]
Thank you! Sorry, I should have read the 'fine' manual instead of skim it.

Constant time Fibonacci:

    def fib x = floor(((((1+sqrt 5)/2)**x)/sqrt 5)+1/2)

    fib 10000

**Comment by robpike** [comment]
Too many parens!

def fib x = floor 1/2 + ((1/2 * 1+sqrt 5)**x) / sqrt 5

You learn the style after a while.

Note that 1/2 is a rational, but 1 / 2 is a division. APL had the divide symbol, which avoided the ambiguity, but then required a special symbol for negative. Ivy does it lexically: -1 vs. - 1.

Also: When the next release comes out, I hope soon (it's gated by mobile), the 'def' word will become 'op' to avoid issues when in base 16.
